ecmd-core/dll/ecmdDllCapi.C

```diff
@@ -5227,24 +5223,23 @@ uint32_t readScandef(const ecmdChipTarget & target, const char* i_ringName, cons
     std::string l_version = "default";
     /* Let's see if we have already looked up this info */
     /* find scandef file */
-    rc = dllQueryFileLocation(target, ECMD_FILE_SCANDEF, l_filePairs, l_version);
+    rc = dllQueryFileLocation(target, ECMD_FILE_SCANDEF, l_fileLocs, l_version);
     if (rc) {
       dllRegisterErrorMsg(rc, "readScandef", "Error occured locating scandef file.\n");
       return rc;
     }
 
-    if (findLatchInCache(l_filePairs, latchHashKey64, ringName, o_latchdata))
+    if (findLatchInCache(l_fileLocs, latchHashKey64, ringName, o_latchdata))
     {
         /* We're done, get out of here */
         return rc;
     }
 
     /* We don't have it already, let's go looking */
-    std::list<std::pair<std::string, std::string> >::iterator l_filePair = l_filePairs.begin();
-    while (l_filePair != l_filePairs.end())
+    for (auto l_fileLoc : l_fileLocs)
```

Comment:
This is going to cause problems with some of the old 32 bit compilers for architectures we support.